stores/governance/src/main/java/com/bloxbean/cardano/yaci/store/governance/processor/DRepRegistrationProcessor.java

```diff
@@ -0,0 +1,104 @@
+    private final DRepRegistrationStorage drepRegistrationStorage;
+
+    @Transactional
+    @EventListener
+    public void handleDRepRegistration(CertificateEvent certificateEvent) {
+        EventMetadata eventMetadata = certificateEvent.getMetadata();
+
+        List<DRepRegistration> dRepRegistrations = new ArrayList<>();
+        for (TxCertificates txCertificates : certificateEvent.getTxCertificatesList()) {
+            String txHash = txCertificates.getTxHash();
+            int index = 0;
+
+            for (var certificate : txCertificates.getCertificates()) {
+                DRepRegistration drepRegistration = switch (certificate.getType()) {
+                    case REG_DREP_CERT -> {
+                        RegDrepCert regDrepCert = (RegDrepCert) certificate;
+                        yield buildDRepRegistration(regDrepCert.getDrepCredential(), regDrepCert.getAnchor(),
+                                regDrepCert.getCoin(), certificate.getType(),
+                                txHash, index, eventMetadata);
+                    }
+                    case UNREG_DREP_CERT -> {
+                        UnregDrepCert unregDrepCert = (UnregDrepCert) certificate;
+                        yield buildDRepRegistration(unregDrepCert.getDrepCredential(), null, unregDrepCert.getCoin().negate(),
+                                certificate.getType(), txHash, index, eventMetadata);
+                    }
+                    case UPDATE_DREP_CERT -> {
+                        UpdateDrepCert updateDrepCert = (UpdateDrepCert) certificate;
+                        yield buildDRepRegistration(updateDrepCert.getDrepCredential(), updateDrepCert.getAnchor(), null,
+                                certificate.getType(), txHash, index, eventMetadata);
+                    }
+                    default -> null;
+                };
+
+                if (drepRegistration != null) {
+                    dRepRegistrations.add(drepRegistration);
+                }
+                index++;
+            }
+        }
+        if (!dRepRegistrations.isEmpty()) {
+            drepRegistrationStorage.saveAll(dRepRegistrations);
+        }
+    }
+
+    private DRepRegistration buildDRepRegistration(Credential drepCredential, Anchor anchor, BigInteger deposit,
+                                                   CertificateType certificateType, String txHash,
+                                                   int certIndex, EventMetadata eventMetadata) {
+        DRepRegistration drepRegistration = DRepRegistration.builder()
+                .txHash(txHash)
+                .certIndex(certIndex)
+                .type(certificateType)
+                .slot(eventMetadata.getSlot())
+                .blockNumber(eventMetadata.getBlock())
+                .blockTime(eventMetadata.getBlockTime())
+                .drepHash(drepCredential.getHash())
+                .drepId(drepCredential.getHash() != null ? DRepId.fromKeyHash(drepCredential.getHash()) : null)
```

Comment:
@Sotatek-HuyLe3a Can you please add a //TODO here ? We need to check if the DRepId is different for script hash. Otherwise, we have to change the method name from  DRepId.fromKeyHash to DRepId.fromCredential